File: SDKs/amazonaws.com/iotsitewise/2019-12-02/python/src/sdk/models/shared/assetmodelcompositemodeldefinition.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class AssetModelCompositeModelDefinition:
    r"""AssetModelCompositeModelDefinition
    Contains a composite model definition in an asset model. This composite model definition is applied to all assets created from the asset model.
    """
    
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    type: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    properties: Optional[Any] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('properties') }})
    

File: SDKs/amazonaws.com/iotsitewise/2019-12-02/java/lib/src/main/java/openapisdk/models/shared/AssetModelCompositeModelDefinition.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * AssetModelCompositeModelDefinition
 * Contains a composite model definition in an asset model. This composite model definition is applied to all assets created from the asset model.
**/
public class AssetModelCompositeModelDefinition {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("description")
    public String description;
    public AssetModelCompositeModelDefinition withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonProperty("name")
    public String name;
    public AssetModelCompositeModelDefinition withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("properties")
    public Object properties;
    public AssetModelCompositeModelDefinition withProperties(Object properties) {
        this.properties = properties;
        return this;
    }
    @JsonProperty("type")
    public String type;
    public AssetModelCompositeModelDefinition withType(String type) {
        this.type = type;
        return this;
    }
}